go/proof.go

```diff
@@ -204,6 +205,14 @@ func (p *ExistenceProof) CheckAgainstSpec(spec *ProofSpec) error {
 	return nil
 }
 
+func keyForComparison(spec *ProofSpec, key []byte) []byte {
```

Comment:
might be nice to add a little doc string explaining the usage of this function, that is to obtain the key used for lexical ordering in non existence proofs because prehashing the key changes the lexical ordering.